folly/FBString.h

```diff
@@ -1010,10 +1010,10 @@ class basic_fbstring {
   typedef typename A::size_type size_type;
   typedef typename A::difference_type difference_type;
```

Comment:
https://en.cppreference.com/w/cpp/named_req/Allocator

`size_type` and `difference_type` are also optional - should these also be using `std::allocator_traits`?

Reply:
We can use std::allocator_traits but as these members are still in std::allocator and are not flagged as deprecated, I think it is not mandatory. Up to you.

Reply:
But they may not exist in other allocator types besides `std::allocator`.

Reply:
True.